nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/util/RecordPathUtils.java

```diff
@@ -39,4 +39,44 @@ public static String getFirstStringValue(final RecordPathSegment segment, final
 
         return stringValue;
     }
+
+    public static String unescapeBackslash(String value) {
+        if (value == null || value.isEmpty()) {
+            return value;
+        }
+        // need to escape characters after backslashes
+        final StringBuilder sb = new StringBuilder();
+        boolean lastCharIsBackslash = false;
+        for (int i = 0; i < value.length(); i++) {
+            final char c = value.charAt(i);
+
```

Comment:
Is this the entire list that has to be escaped?  Maybe you can reference where that list is specified for maintenance.

Reply:
I am sorry, I meant if you could document the code.  That way someone who goes in can tell that you are escaping these specific characters per some specification

Reply:
@ottobackwards I've added Javadocs with reference to antlr parser and explaining the reasons for this transformation.